bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,710 @@
+
+    return Pair.newPair(valueAt(columnId), true);
+  }
+
+  private void clear() {
+    this.maxValue = 0;
+    this.minValue = 0;
+    this.ebM = null;
+    this.bA = null;
+  }
+
+  /**
+   * Serialize this bitmapsliceindex(bsi).
+   * <p>
+   * There is no specification for now: it may change from one java version to
+   * another. This serialization format is compatible with bsi of roaring(Go)
+   * @see <a href="https://github.com/RoaringBitmap/roaring/blob/b32ae1a8cc386adbf7753251b8a580ed37f270a9/roaring64/bsi64.go#L882">bsi64.WriteTo</a>
+   * Current serialization format:
+   * ---
+   * ebM: as standard Roaring64NavigableMap format (with specify legacy/portable format)
+   * loop over bA[]:  as standard Roaring64NavigableMap format (with specify legacy/portable format)
+   * ---
+   *
+   * <p>
+   * Consider calling {@link #runOptimize} before serialization to improve compression.
+   * <p>
+   * The current bsi is not modified.
+   *
+   * @param output the DataOutput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void serialize(DataOutput output) throws IOException {
+    this.ebM.serialize(output);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serialize(output);
+    }
+  }
+
+  public void serialize(ByteBuffer buffer) throws IOException {
+    this.ebM.serializePortable(buffer);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serializePortable(buffer);
+    }
+  }
+
+  /**
+   * Deserialize (retrieve) this bitmapsliceindex(bsi).
+   * <p>
+   * The default serialization format for underlying Roaring64NavigableMap is
+   * {@link Roaring64NavigableMap#SERIALIZATION_MODE_PORTABLE}
+   * <p>
+   * The current bsi is overwritten.
+   *
+   * @param in the DataInput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void deserialize(DataInput in) throws IOException {
+    this.clear();
+
+    // read ebm
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    // set serialization format before deserialize
+    ebm.setSerializationMode(serializationMode);
+    ebm.deserialize(in);
+    this.ebM = ebm;
+
+    List<Roaring64NavigableMap> baList = new ArrayList<>();
+    try (DataInputStream is = new DataInputStream((InputStream) in); ) {
+      while (is.available() > 0) {
+        Roaring64NavigableMap rb = new Roaring64NavigableMap();
+        // set serialization format before deserialize
+        rb.setSerializationMode(serializationMode);
+        rb.deserialize(in);
+        baList.add(rb);
+      }
+    }
+    this.bA = baList.toArray(new Roaring64NavigableMap[0]);
+  }
+
+  public void deserialize(ByteBuffer buffer) throws IOException {
+    this.clear();
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    ebm.setSerializationMode(serializationMode);
+    ebm.deserialize(buffer);
+    this.ebM = ebm;
+    // read back
+    buffer.position(buffer.position() + (int) ebm.serializedSizeInBytes());
```

Comment:
What if `ebm.serializedSizeInBytes()`> Integer.MAX_VALUE?

I see you do a `Roaring64NavigableMap.deserialize(buffer);`, indeed you introduce such a method. I feel uncomfortable about this. There is some related notes in https://stackoverflow.com/questions/4071998/could-bytebuffer-implement-dataoutput-datainput

Should we rather stick with `DataInput`/`DataOutput` for `64` bitmaps?

Reply:
will remove ser/deser using  bytebuffer

Reply:
(As said later, I'm fine keeping them as they were introduced too for Roaring64Bitmap)